pkg/api/scylla/validation/cluster_validation.go

```diff
@@ -99,6 +108,25 @@ func ValidateScyllaClusterSpec(spec *scyllav1.ScyllaClusterSpec, fldPath *field.
 	}
 
 	if spec.ExposeOptions != nil {
+		supportsExposingViaService, err := scyllafeatures.VersionSupports(spec.Version, scyllafeatures.ExposingScyllaClusterViaServiceOtherThanClusterIP)
+		if err != nil {
+			allErrs = append(allErrs, field.InternalError(fldPath.Child("version"), fmt.Errorf("can't determine if version %q supports exposing via Service: %w", spec.Version, err)))
```

Comment:
I agree it can be an issue, but not a big one. This rule is enforced only when user requests exposing, not by default. Images published by Scylla, both concrete versions and rolling tags are parsed correctly - except `latest` but this could be added as a special case here, as both `latest` (oss and enterprise) are enough up-to-date to support what we require here. 
Suggested solution, to move it to runtime wouldn't change anything. We don't have a better way of telling which version of scylla we deploy, before we actually deploy, run and check from the Scylla API. Adding metadata check from image registry would be out of scope of this PR.

Reply:
> as both latest (oss and enterprise) are enough up-to-date to support what we require here.

latest can be arbitrarily old if you have it already pulled

> Suggested solution, to move it to runtime wouldn't change anything.

It allows us to do time consuming / async operations like fetching the image manifest and checking the LABELS. Validation is special as it can't afford to reach to external system or retry because it block the client calls.

Reply:
I changed it to runtime check, even though I consider it a degradation of UX.